### ISSUE TITLE
Documentation search result regression

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/julia#52977]
In 1.10, the search "threading" lists https://docs.julialang.org/en/v1.10/base/multi-threading/ first while in 1.11, it lists https://docs.julialang.org/en/v1.11-dev/NEWS/#Multi-threading-changes, https://docs.julialang.org/en/v1.11-dev/devdocs/locks/#Proper-maintenance-and-care-of-multi-threading-locks, and then https://docs.julialang.org/en/v1.11-dev/devdocs/locks/

It seems pretty clear to me that a manual page with an exact keyword match in it's title should rank first while it currently ranks 5th. I think this is a case-sensitivity bug upstream: https://github.com/JuliaDocs/Documenter.jl/issues/2373

1.10:
![Screenshot from 2024-01-19 09-24-41](https://github.com/JuliaLang/julia/assets/60898866/cd230f19-74cc-495e-ba57-0f6dd23e1fbe)

master:
![Screenshot from 2024-01-19 09-25-01](https://github.com/JuliaLang/julia/assets/60898866/70beae5c-d677-4f04-80ac-cf6d17698bff)

@Hetarth02 changed the Documenter.jl search engine from LunrJs to MinisearchJs in https://github.com/JuliaDocs/Documenter.jl/pull/2172. According to https://hetarth02.hashnode.dev/crafting-an-enhanced-search-experience this seems like a good change overall.

We down streamed those changes into the Julia repo in https://github.com/JuliaLang/julia/pull/47105

**Comment by Hetarth02** [comment]
Yes, I also think this might be a case sensitivity issue and there are PRs in work to fix this. I hope to close them as soon as possible.

**Comment by fingolfin** [comment]
This seems to be solved in current releases. E.g. on the v1.12 pages, if I search for `threading` I set this:

<img width="757" height="750" alt="Image" src="https://github.com/user-attachments/assets/ff11b993-d324-41c3-b662-377d1e210f72" />
